platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/internal/AddressBookIterator.java

```diff
@@ -26,12 +28,13 @@
  */
 public class AddressBookIterator implements Iterator<Address> {
 
-    private final Iterator<Long> orderedNodeIds;
-    private final Map<Long, Address> addresses;
+    private final Iterator<NodeId> orderedNodeIds;
+    private final Map<NodeId, Address> addresses;
 
-    public AddressBookIterator(final Iterator<Long> orderedNodeIds, final Map<Long, Address> addresses) {
-        this.orderedNodeIds = orderedNodeIds;
-        this.addresses = addresses;
+    public AddressBookIterator(
+            @NonNull final Iterator<NodeId> orderedNodeIds, @NonNull final Map<NodeId, Address> addresses) {
```

Comment:
Maybe it does not belong to that issue but I ask myself if `Iterable` instead of `Iterator` would make more sense as a param.

Reply:
This object is never directly instantiated by anybody. `AddressBook` implements `Iterable<Address>`, and uses this helper class to do so.